File: libs/eave-stdlib-py/src/eave/stdlib/github_api/operations.py
from typing import Optional
import pydantic
from .models import FileChange


class Endpoint:
    pass


class GetGithubUrlContent(Endpoint):
    class RequestBody(pydantic.BaseModel):
        url: str

    class ResponseBody(pydantic.BaseModel):
        content: Optional[str]


class CreateGithubResourceSubscription(Endpoint):
    class RequestBody(pydantic.BaseModel):
        url: str

class CreateGitHubPullRequest(Endpoint):
    class RequestBody(pydantic.BaseModel):
        repo_name: str
        repo_owner: str
        repo_id: str
        base_branch_name: str
        branch_name: str
        commit_message: str
        pr_title: str
        pr_body: str
        file_changes: list[FileChange]

    class ResponseBody(pydantic.BaseModel):
        pr_number: int

File: libs/eave-stdlib-ts/src/github-api/operations.ts
import { FileChange } from './models.js';

export type GetGithubUrlContentRequestBody = {
  url: string;
}

export type GetGithubUrlContentResponseBody = {
  content: string | null;
}

export type CreateGithubResourceSubscriptionRequestBody = {
  url: string;
}

export type CreateGitHubPullRequestRequestBody = {
  repoName: string,
  repoOwner: string,
  repoId: string,
  baseBranchName: string,
  branchName: string,
  commitMessage: string,
  prTitle: string,
  prBody: string,
  fileChanges: Array<FileChange>,
}

export type CreateGitHubPullRequestResponseBody = {
  pr_number: number,
}
